### PR TITLE
CVideoPlayer: cannot pause if seek is not supported

### PR DESCRIPTION
**Pull request by djp952**
## Description
CVideoPlayer does not allow for a pause operation to occur if seek is not supported.  The call to SetPlaySpeed will be specifically bypassed by a test to CanSeek().

## Motivation and Context
While developing a PVR client I had noted a while back that if the client indicates it supports pausing but it does not support seeking, the player UI control for Pause was present but inoperable.  I ultimately tracked down the problem to CVideoPlayer::SetSpeed disallowing the operation by testing CanSeek().  

The proposed modification checks both CanSeek() and CanPause() up front and returns early if both are false.  Subsequently if CanSeek() is false, implying that CanPause() is true, and the request is for a pause (DVD_PLAYSPEED_PAUSE) or resume (DVD_PLAYSPEED_NORMAL) operation it will be allowed to go through.

## How Has This Been Tested?
I tested this change on Windows (Win32) using a local build of the master branch.  The resultant binaries (.dll, .pdb) were copied over a recently installed nightly build.  I confirmed the change resolves the specific problem I'm trying to resolve with the PVR Client.

Additionally, I temporarily hard-coded CanSeek() to return false and tested the player with a number of different playback mediums - DVD, Blu-Ray, MP4, WMV and M2TS.  Each media type was paused and resumed successfully while disallowing seek and playback speed changes as expected.

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -3589,12 +3589,18 @@ void CVideoPlayer::SetSpeed(float speed)
   // forward is fine
   if (speed < 0 && IsInMenu())
     return;
-
-  if (!CanSeek())
+ 
+  if(!CanSeek() && !CanPause())
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Meh. Please nclude a space between "if" and "(".

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 2]
```diff
@@ -3589,12 +3589,18 @@ void CVideoPlayer::SetSpeed(float speed)
...
     return;
   
   int iSpeed = static_cast<int>(speed * DVD_PLAYSPEED_NORMAL);
 
+  if(!CanSeek())
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Please nclude a space between "if" and "(".

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 3]
```diff
@@ -3589,12 +3589,18 @@ void CVideoPlayer::SetSpeed(float speed)
...
+  {
+    if((iSpeed != DVD_PLAYSPEED_NORMAL) && (iSpeed != DVD_PLAYSPEED_PAUSE))
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Please nclude a space between "if" and "(".

**Comment by djp952** [comment]
Spaces added

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 4]
```diff
@@ -3589,12 +3589,18 @@ void CVideoPlayer::SetSpeed(float speed)
   // forward is fine
   if (speed < 0 && IsInMenu())
     return;
-
-  if (!CanSeek())
+
```
Comment (hudokkow) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: stray space.

Thanks for your contribution. ping @FernetMenta

**Comment by djp952** [comment]
Stray spaces eliminated like the rabid dogs that they are.

**Comment by FernetMenta** [comment]
thanks. build errors unrelated

**Review by FernetMenta — COMMENTED** [review]
+1

**Comment by ksooo** [comment]
Maybe next time we should at least take our time to squash into a single commit before we merge... This is, how we usually work - eliminate temporary commits, then merge.

**Comment by FernetMenta** [comment]
oops, missed that. sorry

**Comment by djp952** [comment]
Thanks ksooo, I have practiced squashing commits offline so I'll get that right next time.  'preciate it!

**Pull request merged**